From af66a281ee2bcc7cc6c48bb3c8d033d385c12875 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sun, 14 Jun 2020 12:31:00 +0200
Subject: [PATCH] Fix use of base64.decodestring() in tests

Use base64.decodebytes() instead of deprecated/removed .decodestring()
in test_user_data_with_run_instance().  Fixes tests on py3.9.
---
 tests/test_ec2/test_instances.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/test_ec2/test_instances.py b/tests/test_ec2/test_instances.py
index 041bc8c8..1def1734 100644
--- a/tests/test_ec2/test_instances.py
+++ b/tests/test_ec2/test_instances.py
@@ -9,6 +9,7 @@ from nose.tools import assert_raises
 import base64
 import datetime
 import ipaddress
+import sys
 
 import six
 import boto
@@ -765,7 +766,10 @@ def test_user_data_with_run_instance():
     instance_attribute = instance.get_attribute("userData")
     instance_attribute.should.be.a(InstanceAttribute)
     retrieved_user_data = instance_attribute.get("userData").encode("utf-8")
-    decoded_user_data = base64.decodestring(retrieved_user_data)
+    if sys.version_info[0] == 2:
+        decoded_user_data = base64.decodestring(retrieved_user_data)
+    else:
+        decoded_user_data = base64.decodebytes(retrieved_user_data)
     decoded_user_data.should.equal(b"some user data")
 
 
-- 
2.27.0

